app/models/manageiq/providers/microsoft/infra_manager/scanning/job.rb

```diff
@@ -0,0 +1,173 @@
+class ManageIQ::Providers::Microsoft::InfraManager::Scanning::Job < VmScan
+  # Make updates to default state machine to take into account snapshots
+  def load_transitions
+    super.tap do |transitions|
+      transitions.merge!(
+        :start_snapshot     => {'before_scan'               => 'snapshot_create'},
+        :snapshot_complete  => {'snapshot_create'           => 'scanning',
+                                'snapshot_delete'           => 'synchronizing'},
+        :snapshot_delete    => {'scanning'                  => 'snapshot_delete'},
+        :data               => {'snapshot_create'           => 'scanning',
+                                'scanning'                  => 'scanning',
+                                'snapshot_delete'           => 'snapshot_delete',
+                                'synchronizing'             => 'synchronizing',
+                                'finished'                  => 'finished'}
+      )
+    end
+  end
+
+  def before_scan
+    signal(:start_snapshot)
+  end
+
+  def call_snapshot_create
+    _log.info("Enter")
```

Comment:
I vote for a nicer message here. :)

Reply:
@djberg96 while I agree, I would do that in a separate PR.  I am just refactoring here - trying to change the least amount possible and ensure everything still works.